shared/src/event_handling.rs

```diff
@@ -0,0 +1,181 @@
+use anyhow::{Context, Error, Result};
+use ethcontract::contract::{AllEventsBuilder, ParseLog};
+use ethcontract::errors::ExecutionError;
+use ethcontract::{dyns::DynTransport, BlockNumber as Web3BlockNumber, Event as EthcontractEvent};
+use futures::{Stream, StreamExt, TryStreamExt};
+use std::ops::RangeInclusive;
+use web3::Web3;
+
+// We expect that there is never a reorg that changes more than the last n blocks.
+const MAX_REORG_BLOCK_COUNT: u64 = 25;
+// When we insert new trade events into the database we will insert at most this many in one
```

Comment:
nit: Maybe this comment should reference EventStoring now instead of a database.